language/bytecode-verifier/src/verifier.rs

```diff
@@ -234,105 +231,6 @@ pub fn verify_script_dependency_map(
-                    idx,
-                    StatusCode::MISSING_DEPENDENCY,
-                ))
-            }
-            Some(vm_native_function) => {
-                // check parameters
-                let def_params = native_function_definition_view.parameters();
-                let native_params = match vm_native_function.parameters(Some(module_view))? {
-                    None => {
-                        return Err(verification_error(
-                            IndexKind::FunctionHandle,
-                            idx,
-                            StatusCode::TYPE_MISMATCH,
-                        ))
-                    }
-                    Some(sig) => sig,
-                };
-                if def_params != &native_params {
-                    return Err(verification_error(
-                        IndexKind::FunctionHandle,
-                        idx,
-                        StatusCode::TYPE_MISMATCH,
-                    ));
-                }
-
-                // check return_
-                let def_return_ = native_function_definition_view.return_();
-                let native_return_ = match vm_native_function.return_(Some(module_view))? {
-                    None => {
-                        return Err(verification_error(
-                            IndexKind::FunctionHandle,
-                            idx,
-                            StatusCode::TYPE_MISMATCH,
-                        ))
-                    }
-                    Some(sig) => sig,
-                };
-                if def_return_ != &native_return_ {
-                    return Err(verification_error(
-                        IndexKind::FunctionHandle,
-                        idx,
-                        StatusCode::TYPE_MISMATCH,
-                    ));
-                }
-
-                // check type parameters
-                let def_type_parameters = native_function_definition_view.type_parameters();
-                let native_type_parameters =
-                    match vm_native_function.type_parameters(Some(module_view))? {
-                        None => {
-                            return Err(verification_error(
-                                IndexKind::FunctionHandle,
-                                idx,
-                                StatusCode::TYPE_MISMATCH,
-                            ))
-                        }
-                        Some(t_params) => t_params,
-                    };
-                if def_type_parameters != &native_type_parameters {
-                    return Err(verification_error(
-                        IndexKind::FunctionHandle,
-                        idx,
-                        StatusCode::TYPE_MISMATCH,
-                    ));
-                }
-            }
-        }
-    }
-    Ok(())
-}
-
-// TODO: native structs have been partially removed. Revisit.
-fn verify_native_structs(module_view: &ModuleView<VerifiedModule>) -> VMResult<()> {
```

Comment:
same here, I don't feel like this should go away

Reply:
this follows the same logic as the functions. It's a publish and load time.
However structs in this PR are shortcircuit'ed in a horrible way and we should talk about where to fail them.
In other words there is nothing a verifier - or anything that is not the runtime that owns the natives - can do or decide upon. 
To put it another way, the verifier should not "version" if natives change.

Thoughts?

Reply:
This makes sense. I think what is there in the new `check_natives` is good for now